File: Service/ProjectManagement.Solution/ProjectManagement.Services.Contracts/ITaskService.cs
using ProjectManagement.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjectManagement.Services.Contracts
{
    public interface ITaskService : IDisposable
    {
        ICollection<TaskModel> GetAllTasks();

        TaskModel GetTaskById(int taskId);

        bool CreateTask(TaskModel taskModel);

        bool EditTask(TaskModel taskModel);

        bool DeleteTask(int taskId);

        bool EndTask(int taskId);
    }
}

File: UI/ProjectManagementApp/src/app/ITaskService.ts
import { Observable } from 'rxjs';
import { Task } from './Models/Task';
import { CreateTaskModel } from './Models/CreateTaskModel';
import { EditTaskModel } from './Models/EditTaskModel';



export abstract class ITaskService {
    abstract getAllTask(): Observable<Task[]>;

    abstract getTaskById(taskId: number): Observable<Task>;

    abstract createTask(createTaskModel: CreateTaskModel): Observable<CreateTaskModel>;

    abstract editTask(editTaskComponent: EditTaskModel): Observable<EditTaskModel>;
}